### PR TITLE
ci: single source of truth for Envoy Docker base image.

### PR DESCRIPTION
**Pull request by htuch**
Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

File: ci/envoy_build_sha.sh [hunk 1]
```diff
@@ -1,3 +1 @@
-# If you edit the SHA here you must also edit the SHA in .circleci/config.yml.
-
-ENVOY_BUILD_SHA=114e24c6fd05fc026492e9d2ca5608694e5ea59d
+ENVOY_BUILD_SHA=$(grep lyft/envoy-build .circleci/config.yml | sed -e 's#.*lyft/envoy-build:\(.*\)#\1#' | head -n 1)
```
Comment (mattklein123) [inline ci/envoy_build_sha.sh]: Is there any way to see if someone put different values inside config.yml? If that's too annoying nbd but thought I would ask.

Comment (htuch) [reply ci/envoy_build_sha.sh]: You just want to validate the consistency of all the hashes that appear in that file?

Comment (mattklein123) [reply ci/envoy_build_sha.sh]: Yeah they should all be the same.

**Comment by mattklein123** [comment]
Looks like there still might be some OSX issues on http/2 integration test? cc @zuercher @alyssawilk/ @htuch I assume this branch is up to date otherwise?

**Comment by htuch** [comment]
Yes, was up-to-date otherwise.

**Pull request merged**